src/main/java/gregtech/common/metatileentities/storage/CraftingRecipeLogic.java

```diff
@@ -314,17 +317,26 @@ public IRecipe getCachedRecipe() {
         return this.cachedRecipeData.getRecipe();
     }
 
-    public void update() {
+    @Override
+    public void detectAndSendChanges(boolean init) {
+        super.detectAndSendChanges(init);
+        this.collectAvailableItems();
```

Comment:
I'm a little concerned about moving `collectAvailableItems()` to be called every tick, but i'm not totally sure how to handle it better yet.

Reply:
This would only occur inside the GUI; therefore, I'm not too terribly worried about this. Also, yes, I have no idea how I would actually detect changes within any possible inventory. Maybe I'll just do this every other tick.